synapse/storage/room.py

```diff
@@ -500,10 +500,12 @@ def get_all_new_public_rooms(txn):
 
     @defer.inlineCallbacks
     def block_room(self, room_id, user_id):
-        yield self._simple_insert(
+        yield self._simple_upsert(
```

Comment:
has this just been broken since we did the native upsert thing?

Reply:
No? I don't really follow.

Reply:
well, why are you changing it?

Reply:
So that it doesn't fail when we call it twice, as per comment in admin code

Reply:
heh, I'd failed to spot that you were changing insert to upsert. makes much more sense now.